docs/RFCS/20211130_node_id_refactoring.md

```diff
@@ -0,0 +1,115 @@
+- Feature Name: Refactoring KV and SQL node iDs
+- Status: draft
+- Start Date: 2021-11-31
+- Authors: Radu Berinde
+- RFC PR: (PR # after acceptance of initial draft)
+- Cockroach Issue: (one or more # from the issue tracker)
+
+
+# Background
```

Comment:
There's a sad piece of complexity I want to pile onto all of this. Hopeful we can make that complexity superficial only. Today and in this RFC there's an assumption that we're content to have all KV nodes use their node ID as the one and only ID for ~everything and then in SQL we'll use the `SQLInstanceID`. The `SQLInstanceID` used in tenant pods is leased to always provide small (<16 bit) values. The problem with node IDs is that they can get big. This breaks `unique_rowid`. The fix I'm interested in is having the sql nodes also grab an instance ID using the same protocol we added and use that for `unique_rowid`. 

Anyway, if we do that, do we want a concept of a leased id as something different from either of these for use just in `unique_rowid`? 

https://github.com/cockroachdb/cockroach/issues/47470